src/Uno.UI/UI/Xaml/Controls/Border/Border.cs

```diff
@@ -100,22 +100,40 @@ public UIElement Child
 				// disable the property-based propagation here to support the case where the Parent property is overridden to simulate
 				// a different inheritance hierarchy, as is done for some controls with native styles.
 				FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext,
-				(s, e) => ((Border)s)?.OnChildChanged((UIElement)e.OldValue, (UIElement)e.NewValue)
+				(DependencyObject s, DependencyPropertyChangedEventArgs e) => ((Border)s)?.OnChildChanged((UIElement)e.OldValue, (UIElement)e.NewValue),
+				OnCoerceChild
 			)
 		);
 
-	private void OnChildChanged(UIElement oldValue, UIElement newValue)
+	private static object OnCoerceChild(DependencyObject dependencyObject, object baseValue, DependencyPropertyValuePrecedences precedence)
 	{
-		ReAttachChildTransitions(oldValue, newValue);
+		// Until we do the breaking change (#8339), Child is a Dependency property in Uno Platform.
+		// This means that setting Child to the same reference does not trigger PropertyChanged, 
+		// which is a problem, as WinUI always removes the current child and adds it again,
+		// even when dealing with the exact same reference. This also triggers measure/arrange
+		// on the child. To work around this, we force the update in coerce here. 
```

Comment:
In WinUI, Child is a DP, but only `Child` is exposed publicly. The static `ChildProperty` is not exposed.

```cpp
    // Microsoft.UI.Xaml.Controls.Border.Child
    {
        KnownPropertyIndex::Border_Child, // Index name
        KnownTypeIndex::UIElement, // Property type
        KnownTypeIndex::Border, // Declaring type
        KnownTypeIndex::Border, // Target type
        static_cast<MetaDataPropertyInfoFlags>(static_cast<UINT>(MetaDataPropertyInfoFlags::AffectMeasure) | static_cast<UINT>(MetaDataPropertyInfoFlags::IsPublic) | static_cast<UINT>(MetaDataPropertyInfoFlags::IsPropMethodCall) | static_cast<UINT>(MetaDataPropertyInfoFlags::RequiresMultipleAssociationCheck) | static_cast<UINT>(MetaDataPropertyInfoFlags::IsVisualTreeProperty)), // Property flags
    },
```